Userland/Services/WindowServer/WindowManager.cpp

```diff
@@ -1211,7 +1223,17 @@ void WindowManager::event(Core::Event& event)
             return;
         }
 
-        if (MenuManager::the().current_menu()) {
+        if (key_event.type() == Event::KeyDown && key_event.key() == Key_Super) {
+            m_previous_event_was_super_keydown = true;
```

Comment:
After `m_previous_event_was_super_keydown` is set to `true`, it stays `true` even if the next event is e.g a `MouseDown` event. That makes the value a lie, and means that any number of unrelated mouse events can happen in between super keydown/keyup.

If you look at 7ee6c66ee95ad151c24d34c259464db03fb7ea6b where this functionality was originally implemented, notice that `m_previous_event_is_key_down_logo` was cleared upon any mouse event other than `MouseMove`.

Reply:
Oh, okay.. I've added that and I'll just test it before pushing. Thank you :^)